addons/website_sale_coupon/models/sale_order.py

```diff
@@ -93,3 +113,29 @@ def _gc_abandoned_coupons(self, *args, **kwargs):
             coupon.sales_order_id.applied_coupon_ids -= coupon
         coupon_to_reset.write({'state': 'new'})
         coupon_to_reset.mapped('sales_order_id').recompute_coupon_lines()
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    def unlink(self):
+        reward_lines = self.filtered(lambda line: line.is_reward_line)
+        status = request.session.get('pending_coupon') or {}
```

Comment:
How can this work? `pending_coupon` shouldn't exist anymore if the coupon reward line was created (and it is since we are now deleting it?)

Reply:
Because of @sbuhl 's request, `pending_coupon` stays unless it is explicitly deleted from the cart. This allows the coupon to be kept even if the cart temporarily doesn't allow the coupon to be applied.